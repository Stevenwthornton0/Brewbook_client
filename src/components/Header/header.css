.header {
    width: 100%;
}

.header_container {
    width: 95%;
    margin-left: auto;
    margin-right: auto;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}


.header_logged-out, .header_logged-in {
    margin-top: 1.2em;
    font-size: 20px;
    color: #fedd02
}

a {
    color: #fedd02;
    text-decoration: none;
}

nav {
    width: 100%;
}

.logo {
    color: #fedd02;
}

.logo > h1 {
    font-size: 26px;
    margin: 20px 0;
}

@media screen and (max-width: 500px) {
    .header_container {
        width: 90%;
    }
    
    /* .Hyph {
        display: none;
    } */

    .header_logged-out > a {
        font-size: 18px;
    }
}
