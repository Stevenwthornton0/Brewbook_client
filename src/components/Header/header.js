import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Hyph } from '../../Utils/Utils';
import './header.css';

export default class Header extends Component {

    handleLogoutClick = () => {
        TokenService.clearAuthToken()
        window.location.reload()
    }

    renderLogoutLink() {
        return (
            <div className='header_logged-in'>
                <Link
                    onClick={this.handleLogoutClick}
                    to='/'>
                    Logout
                </Link>     
            </div>
        )
    }

    renderLoginLink() {
        return (
            <div className='header_logged-out'>
                <Link to='/register'>
                    Register
                </Link>
                <Hyph/>
                <Link to='/login'>
                    Login
                </Link>
            </div>
        )
    }

    render() {
        return (
            <nav className='header'>
                <div className='header_container'>
                    <Link to='/' className='logo'>
                        <h1>
                            <FontAwesomeIcon className='logo' icon='beer' />
                            {' '}
                            brewbook
                        </h1>
                    </Link>
                    {TokenService.hasAuthToken()
                        ? this.renderLogoutLink()
                        : this.renderLoginLink()}
                </div>
            </nav>
        )
    }
}

// displays navigation to registration and login/logout