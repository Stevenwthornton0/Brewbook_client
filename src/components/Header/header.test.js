import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import Header from './header';

Enzyme.configure({ adapter: new Adapter() })

describe(`header smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<Header />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})