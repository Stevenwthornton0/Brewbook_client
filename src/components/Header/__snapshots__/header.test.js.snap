// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header smoke test renders without crashing 1`] = `
<nav
  className="header"
>
  <div
    className="header_container"
  >
    <Link
      className="logo"
      to="/"
    >
      <h1>
        <FontAwesomeIcon
          border={false}
          className="logo"
          fixedWidth={false}
          flip={null}
          icon="beer"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         
        brewbook
      </h1>
    </Link>
    <div
      className="header_logged-out"
    >
      <Link
        to="/register"
      >
        Register
      </Link>
      <Hyph />
      <Link
        to="/login"
      >
        Login
      </Link>
    </div>
  </div>
</nav>
`;
